include/boost/config/select_compiler_config.hpp

```diff
@@ -92,8 +92,12 @@
 //  MPW MrCpp or SCpp
 #   define BOOST_COMPILER_CONFIG "boost/config/compiler/mpw.hpp"
 
+#elif defined(__IBMCPP__) && defined(__COMPILER_VER__)
```

Comment:
I'm pretty sure those macros are common to all IMB C++ compliers, not just z/OS, we need a better check here, otherwise it will break IBM C++ on other platforms.